orekit/tests/validate_frames.py

```diff
@@ -0,0 +1,307 @@
+
+# All interesting 3D directions
+R_SET, V_SET = [list(product([-1, 0, 1], repeat=3)) for _ in range(2)]
+
+# Retrieve celestial bodies from orekit
+Sun_orekit = CelestialBodyFactory.getSun()
+Mercury_orekit = CelestialBodyFactory.getMercury()
+Venus_orekit = CelestialBodyFactory.getVenus()
+Earth_orekit = CelestialBodyFactory.getEarth()
+Moon_orekit = CelestialBodyFactory.getMoon()
+Mars_orekit = CelestialBodyFactory.getMars()
+Jupiter_orekit = CelestialBodyFactory.getJupiter()
+Saturn_orekit = CelestialBodyFactory.getSaturn()
+Uranus_orekit = CelestialBodyFactory.getUranus()
+Neptune_orekit = CelestialBodyFactory.getNeptune()
+
+# Name of the bodies
+BODIES_NAMES = [
+    "Sun",
+    "Mercury",
+    "Venus",
+    "Earth",
+    "Moon",
+    "Mars",
+    "Jupiter",
+    "Saturn",
+    "Uranus",
+    "Neptune",
+]
+
+# orekit: bodies, inertial and fixed frames
+OREKIT_BODIES = [
+    Sun_orekit,
+    Mercury_orekit,
+    Venus_orekit,
+    Earth_orekit,
+    Moon_orekit,
+    Mars_orekit,
+    Jupiter_orekit,
+    Saturn_orekit,
+    Uranus_orekit,
+    Neptune_orekit,
+]
+OREKIT_FIXED_FRAMES = [body.getBodyOrientedFrame() for body in OREKIT_BODIES]
+
+# poliastro: bodies, inertial and fixed frames
+POLIASTRO_BODIES = [
+    Sun_poliastro,
+    Mercury_poliastro,
+    Venus_poliastro,
+    Earth_poliastro,
+    Moon_poliastro,
+    Mars_poliastro,
+    Jupiter_poliastro,
+    Saturn_poliastro,
+    Uranus_poliastro,
+    Neptune_poliastro,
+]
+POLIASTRO_ICRS_FRAMES = [
+    HCRS_poliastro,
+    MercuryICRS_poliastro,
+    VenusICRS_poliastro,
+    GCRS_poliastro,
+    MoonICRS_poliastro,
+    MarsICRS_poliastro,
+    JupiterICRS_poliastro,
+    SaturnICRS_poliastro,
+    UranusICRS_poliastro,
+    NeptuneICRS_poliastro,
+]
+POLIASTRO_FIXED_FRAMES = [
+    SunFixed_poliastro,
+    MercuryFixed_poliastro,
+    VenusFixed_poliastro,
+    ITRS_poliastro,
+    MoonFixed_poliastro,
+    MarsFixed_poliastro,
+    JupiterFixed_poliastro,
+    SaturnFixed_poliastro,
+    UranusFixed_poliastro,
+    NeptuneFixed_poliastro,
+]
+
+
+# Collect both API data in two dictionaries
+OREKIT_BODIES_AND_FRAMES = dict(
+    zip(BODIES_NAMES, zip(OREKIT_BODIES, OREKIT_FIXED_FRAMES))
+)
+POLIASTRO_BODIES_AND_FRAMES = dict(
+    zip(
+        BODIES_NAMES,
+        zip(POLIASTRO_BODIES, POLIASTRO_ICRS_FRAMES, POLIASTRO_FIXED_FRAMES),
+    )
+)
+
+
+# Macros for J2000 and ICRF frame from Orekit API
+J2000_OREKIT = AbsoluteDate.J2000_EPOCH
+ICRF_FRAME_OREKIT = FramesFactory.getICRF()
+GCRF_FRAME_OREKIT = FramesFactory.getGCRF()
+ITRF_FRAME_OREKIT = FramesFactory.getITRF(IERSConventions.IERS_2010, False)
+
+
+# Some of tests are marked as XFAIL since Orekit implements the data from IAU
+# WGCCRE 2009 report while poliastro uses IAU WGCCRE 2015 one
+
+
+@pytest.mark.parametrize("r_vec", R_SET)
+@pytest.mark.parametrize("v_vec", V_SET)
+@pytest.mark.parametrize(
+    "body_name",
+    [
+        "Sun",
+        pytest.param("Mercury", marks=pytest.mark.xfail),
```

Comment:
Do we know why these tests fail? If not, we could add a comment like

```suggestion
        pytest.param("Mercury", marks=pytest.mark.xfail),  # reason unknown
```

Reply:
Sure! No problem. All of them are due to the fact that poliastro implements the WGCCRE 2015 report data while orekit is built on top WGCCRE 2009 one.